metadata/src/main/resources/common/metadata/PartitionRecord.json

```diff
@@ -17,9 +17,9 @@
   "apiKey": 3,
   "type": "metadata",
   "name": "PartitionRecord",
-  "validVersions": "0-1",
-  // Version 1 implements Eligiable Leader Replicas and LastKnownELR as described in KIP-966.
-  //
+  // Version 1 replaces Replicas with Assignment
+  // Version 2 implements Eligiable Leader Replicas and LastKnownELR as described in KIP-966.
+  "validVersions": "0-2",
```

Comment:
These comments are backwards: Version 1 is ELR, Version 2 JBOD.

Also, we are adding Assignment, but we are not actually using it yet, so I would be careful about stating "replaces Replicas with Assignment".  I would consider stating `Version 2 adds Assignment as a future replacement for Replicas once enabled via MetadaatVersion.isDirectoryAssignmentSupported()`